### PR TITLE
Removes Mecha

### PR DESCRIPTION
**Pull request by Kurgis**
## About The Pull Request

Removes the unused Mecha module, including some related other files such as Mecha Circuits which weren't unincluded. Changes naming convention across a lot of files to match with Baystation for easier readability. Removes Mecha Jetpack code from jetpack.dm. Mecha2 icons file renamed to be the same as the modules file it's used for, same has been done to the mecha sound files. Edits a dungeon room to no longer spawn a non-existent mecha object.

## Why It's Good For The Game

Mecha code was trash, and we can be alleviated to finally be rid of it all.

## Changelog
:cl:
code: The unincluded mecha module has been fully removed from the codebase, along with changing many naming conventions used in code to be closer to Baystation. This should have no effect in-game.
/:cl:

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -409,70 +409,70 @@
...
-			playsound(user, 'sound/mecha/mechturn.ogg', 20, 1)
+			playsound(user, 'sound/mechs/mechturn.ogg', 20, 1)
 			cooldown = world.time
 			return
 	..()
 
 /obj/item/toy/prize/ripley
 	name = "toy ripley"
-	desc = "Mini-Mecha action figure! Collect them all! 1/11."
+	desc = "Mini-exosuit action figure! Collect them all! 1/11."
```
Comment (ACCount12) [inline code/game/objects/items/toys.dm:426]: I see no reason to touch toys, and Mini-Mecha sounds better.

File: code/__DEFINES/research.dm [hunk 2]
```diff
@@ -45,7 +45,7 @@
 #define CAT_STOCKPARTS  "Stock Parts"
 #define CAT_PROSTHESIS  "Prosthesis"
 #define CAT_MINING      "Mining"
-#define CAT_MECHA       "Exosuits"
+#define CAT_MECH       "Exosuits"
```
Comment (ACCount12) [inline code/__DEFINES/research.dm:48]: Alignment is off.

File: code/modules/mechs/_mech_defines.dm [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-#define MECHA_WRECKAGE_ICON					'icons/mecha2/mech_wreckage.dmi'
-#define MECHA_EQUIPMENT_ICON				'icons/mecha2/mech_equipment.dmi'
-#define MECHA_HUD_ICON						'icons/mecha2/mech_hud.dmi'
+#define MECH_WEAPON_OVERLAYS_ICON			'icons/mechs/mech_weapon_overlays.dmi'
+#define MECH_DECALS_ICON					'icons/mechs/mech_decals.dmi'
+#define MECH_PARTS_HELD_ICON				'icons/mechs/mech_parts_held.dmi'
+#define MECH_PARTS_ICON						'icons/mechs/mech_parts.dmi'
+#define MECH_WRECKAGE_ICON					'icons/mechs/mech_wreckage.dmi'
+#define MECH_EQUIPMENT_ICON					'icons/mechs/mech_equipment.dmi'
+#define MECH_HUD_ICON						'icons/mechs/mech_hud.dmi'
```
Comment (ACCount12) [inline code/modules/mechs/_mech_defines.dm:7]: Why are defines renamed to `MECH` when everything else is renamed to `exosuit`?

Comment (Kurgis) [reply code/modules/mechs/_mech_defines.dm:7]: Baystation uses "MECH" as the beginning of their defines, while they have "exosuit" as their variables.

**Pull request merged**